src/support/rand.c

```diff
@@ -155,7 +155,8 @@ __wt_random(WT_RAND_STATE volatile *rnd_state) WT_GCC_FUNC_ATTRIBUTE((visibility
      * of the random state so we can ensure that the calculation operates on the state consistently
      * regardless of concurrent calls with the same random state.
      */
-    WT_ACQUIRE_READ_WITH_BARRIER(rnd, *rnd_state);
+    rnd = *rnd_state;
```

Comment:
So would this have worked if we just left it as is? Adding an additional atomic load.